api/src/main/java/com/github/eyefloaters/console/api/service/KafkaClusterService.java

```diff
@@ -105,33 +110,75 @@ public static Optional<Kafka> findCluster(SharedIndexInformer<Kafka> kafkaInform
                 .findFirst();
     }
 
-    public static Stream<ListenerStatus> externalListeners(Kafka kafka) {
-        return Optional.ofNullable(kafka.getStatus().getListeners())
-                .map(Collection::stream)
-                .orElseGet(Stream::empty)
-                .filter(l -> isExternalListener(kafka, l));
+    public static Optional<ListenerStatus> consoleListener(Kafka kafka) {
+        return kafka.getSpec().getKafka().getListeners().stream()
+            .filter(listener -> !KafkaListenerType.INTERNAL.equals(listener.getType()))
+            .filter(KafkaClusterService::supportedAuthentication)
+            .sorted((l1, l2) -> Integer.compare(listenerSortKey(l1), listenerSortKey(l2)))
+            .findFirst()
+            .map(listener -> listenerStatus(kafka, listener));
     }
 
-    public static boolean isExternalListener(Kafka kafka, ListenerStatus listener) {
-        return kafka.getSpec()
-                .getKafka()
-                .getListeners()
-                .stream()
-                .filter(sl -> sl.getName().equals(listener.getName()))
-                .map(GenericKafkaListener::getType)
-                .filter(Objects::nonNull)
-                .anyMatch(Predicate.not(KafkaListenerType.INTERNAL::equals));
+    static boolean supportedAuthentication(GenericKafkaListener listener) {
+        Optional<KafkaListenerAuthentication> listenerAuth = Optional.ofNullable(listener.getAuth());
+        String authType = listenerAuth.map(auth -> auth.getType()).orElse("");
+
+        if (authType.isBlank()) {
+            return true;
+        } else if (KafkaListenerAuthenticationOAuth.TYPE_OAUTH.equals(authType)) {
+            return true;
+        } else if (KafkaListenerAuthenticationCustom.TYPE_CUSTOM.equals(authType)) {
+            return listenerAuth
+                .map(KafkaListenerAuthenticationCustom.class::cast)
```

Comment:
Can't you use an `instanceof`?

Reply:
We could add a `filter(KafkaListenerAuthenticationCustom.class::isInstance)` as the first step of this chain. It could be a safety check, but at this point after checking the type is `custom` we should be safe to cast it. The only concern would be if in the future Strimzi has more than one subclass of `KafkaListenerAuthentication` that reports a `custom` type.

Reply:
I think I misunderstood the line being questioned - I agree that the else/if statements would be better with `instanceof`s and I'll adjust for that.